nucypher/cli/main.py

```diff
@@ -452,150 +470,140 @@ def ursula(click_config,
-                   value=value,
-                   duration=duration,
-                   start_period=start_period,
-                   end_period=end_period))
+        start_period = ursula_config.miner_agent.get_current_period()
+        end_period = start_period + duration
 
-        raise NotImplementedError
+        if not force:     # Review
+            click.echo("""
+    
+            | Staged Stake |
+    
+            Node: {address}
+            Value: {value}
+            Duration: {duration}
+            Start Period: {start_period}
+            End Period: {end_period}
+    
+            """.format(address=checksum_address,
+                       value=value,
+                       duration=duration,
+                       start_period=start_period,
+                       end_period=end_period))
+
+        miner = Miner(is_me=True,
+                      checksum_address=ursula_config.checksum_public_address,
+                      blockchain=ursula_config.blockchain)
+
+        result = miner.initialize_stake(amount=value, lock_periods=duration)
+        for tx_name, txhash in result.items():
+            click.secho(f'{tx_name} .......... {txhash}')
+        else:
+            click.secho('Successfully transmitted stake initialization transactions', fg='green')
+        return
 
     elif action == 'confirm-activity':
         """Manually confirm activity for the active period"""
         stakes = ursula_config.miner_agent.get_all_stakes(miner_address=checksum_address)
         if len(stakes) == 0:
             raise RuntimeError("There are no active stakes for {}".format(checksum_address))
         ursula_config.miner_agent.confirm_activity(node_address=checksum_address)
+        return
 
-    elif action == 'divide':
+    elif action == 'divide-stake':
         """Divide an existing stake by specifying the new target value and end period"""
 
-        stakes = ursula_config.miner_agent.get_all_stakes(miner_address=checksum_address)
+        stakes = list(ursula_config.miner_agent.get_all_stakes(miner_address=checksum_address))
         if len(stakes) == 0:
             raise RuntimeError("There are no active stakes for {}".format(checksum_address))
 
-        if not index:
+        if index is None:
             for selection_index, stake_info in enumerate(stakes):
                 click.echo("{} ....... {}".format(selection_index, stake_info))
             index = click.prompt("Select a stake to divide", type=click.INT)
 
-        target_value = click.prompt("Enter new target value", type=click.INT)
-        extension = click.prompt("Enter number of periods to extend", type=click.INT)
+        if not value:
+            target_value = click.prompt("Enter new target value", type=click.INT)
+        else:
+            target_value = value
 
-        click.echo("""
-        Current Stake: {}
+        if not duration:
+            extension = click.prompt("Enter number of periods to extend", type=click.INT)
```

Comment:
Perhaps reiterate that a period is equivalent to one day?

Reply:
Sounds good, having a more verbose message here. What do you think about renaming `--duration` to `--periods`?